### PR TITLE
[#951] Publish all release NPM and Python packages to NPM and PyPi

### PR DESCRIPTION
**Pull request by vlad-tokarev**
#951

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -14,11 +14,33 @@
...
+        mo = re.search(VSRE, file_content, re.M)
+        if mo:
+            return mo.group(1)
+        else:
+            raise RuntimeError("Unable to find version string in %s." % (file_content,))
+
+
 setup(
-    name='legion_mlflow_runner',
+    name='legion_mlflow_runner-test',
```
Comment (mcdoker18) [inline setup.py]: Please, fix the name `legion_mlflow_runner` -> `legion-mlflow-runner`

File: setup.py [hunk 2]
```diff
@@ -34,11 +56,9 @@
         'console_scripts': ['legion-mlflow-runner=mlflowrunner.runner:main'],
     },
     install_requires=[
-        # TODO: fix dependencies when we will publish the legion packages to pypi repo
-        # TODO: consider Pipenv usage
-        'legion-sdk @ git+https://github.com/legion-platform/legion.git@1.0.0-rc18#egg=legion-sdk&subdirectory=legion/sdk',
+        f'legion-sdk=={extract_version()}',
```
Comment (mcdoker18) [inline setup.py]: I think It will not work with a dev version.

File: scripts/Jenkinsfile [hunk 3]
```diff
@@ -43,13 +45,18 @@ pipeline {
...
             param_helm_repository = "${params.HelmRepository}"
+            param_pypi_repository = "${params.PyPiRepo}"
             param_docker_registry = "${params.DockerRegistry}"
             param_docker_hub_registry = "${params.DockerHubRegistry}"
             param_git_deploy_key = "${params.GitDeployKey}"
             ///Job parameters
-            updateVersionScript = "scripts/update_version_id.py"
-            sharedLibPath = "pipelines/legionPipeline.groovy"
+            updateVersionScript = "legion-cicd/tools/update_version_files.py"
+            pythonAlpinePath = "legion-cicd/containers/python-alpine"
```
Comment (mcdoker18) [inline scripts/Jenkinsfile]: I don't like that all our build pipelines know the paths of `ci-cd` repository. `legionPipeline.groovy` is an exception.
Why we can't hardcode these parameters in the `legionPipeline.groovy` script?

File: scripts/Jenkinsfile [hunk 4]
```diff
@@ -170,6 +185,26 @@ pipeline {
...
+                                im.inside{
+                                    legion.publishPythonPackage("./", env.param_pypi_repository)
+                                }
+                            }
+                            else {
+                                if (!env.param_stable_release){
+                                    print("Publishing packages to public registry skipped – not stable release")
+                                }
+                                else{
+                                    print("Publishing packages to public registry skipped – publishing option is disabled")
```
Comment (mcdoker18) [inline scripts/Jenkinsfile]: We should push in the nexus for every build and pypi\npm\... for every *stable* release.

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

**Review by mcdoker18 — APPROVED** [review]

**Review by kirillmakhonin — COMMENTED** [review]
I guess we should not publish internal domains

**Review by maxkochubey — APPROVED** [review]

**Comment by vlad-tokarev** [comment]
> I guess we should not publish internal domains

What exactly do you mean? Give a reference plz

**Pull request merged**